src/firmware_new/boards/frankie_v1/io/drivetrain.h

```diff
@@ -1,29 +1,88 @@
 #pragma once
 
+#include "firmware/shared/math/matrix.h"
 #include "firmware_new/boards/frankie_v1/io/drivetrain_unit.h"
 
+typedef struct Drivetrain Drivetrain_t;
 /**
  * Initialize the IO layer for the Drivetrain
  *
  * NOTE: This does not take ownership of the given drive units. The driveunit's must
  *       remain valid until this function is called again.
  *
- * @param [in] front_left_drive_unit
- * @param [in] front_right_drive_unit
- * @param [in] back_left_drive_unit
- * @param [in] back_right_drive_unit
+ * @param front_left_drive_unit [in]
+
+ * @param front_right_drive_unit [in]
+
+ * @param back_left_drive_unit [in]
+
+ * @param back_right_drive_unit [in]
+
+ * @param robot_wheel_geometry [in] Wheel matrix representing the geometric
+ characteristics of the drivetrain. This includes the angle and placement of wheels.
+ */
+DrivetrainUnit_t* io_drivetrain_init(DrivetrainUnit_t* front_left_drive_unit,
+                                     DrivetrainUnit_t* front_right_drive_unit,
+                                     DrivetrainUnit_t* back_left_drive_unit,
+                                     DrivetrainUnit_t* back_right_drive_unit,
+                                     Matrix robot_wheel_geometry);
+
+/*
+ * Destroy the Drivetrain
+ *
+ * @param drivetrain [in] Drivetrain to destroy
+ */
+void io_drivetrain_destroy(Drivetrain_t* drivetrain);
+
+/**
+ * Calculates equivalent wheel speeds that will result in the robot having the input local
+ * speeds.
+ *
+ *           ___        __
+ *  Wheel_0 = | A0  B0  C0 |   | Vx |
+ *  Wheel_1 = | A1  B1  C1 |   | Vy |
+ *  Wheel_2 = | A2  B2  C3 | * | W  |
+ *  Wheel_3 = | A3  B3  C3 |   |_  _|
+ *            |__        __|
+ *
+ * The speed of each wheel can be determined through the inverse matrix of the forwards
+ * kinematic matrix of the robot The matrix defined about is generated from a matrix
+ * pseud-inverse and under-determined and therefore there are infinite possibilities of
+ * what it could be.
+ *
+ *  *
+ *                    Local X
+ *                       ^
+ *                       |
+ *          0  __________|___________  1
```

Comment:
Assuming the numbers 0-3 are the "wheel numbers", they don't match our current convention (goes 0-3 in counter-clockwise order)